src/prefect/utilities/storage.py

```diff
@@ -102,15 +105,37 @@ def extract_flow_from_module(module_str: str, flow_name: str = None) -> "Flow":
     """
 
     # load the module
-    module = importlib.import_module(module_str)
+    module_parts = module_str.split(":", 2)
+
+    if len(module_parts) == 2 and flow_name is not None:
+        raise ValueError(
+            "Provide either `module_str` without an attribute specifier or remove `flow_name`."
+        )
+
+    module_name, flow_name = (
+        module_parts if len(module_parts) == 2 else [module_str, flow_name or ""]
+    )
+
+    module = importlib.import_module(module_name)
 
     # if flow_name is specified, grab it from the module
     if flow_name:
-        return getattr(module, flow_name)
+        attr = attrgetter(flow_name)(module)
+
+        if callable(attr):
```

Comment:
Why support callable objects? Is this something you need? I'm fine with adding it, but would like to hear a use case beforehand.

Reply:
Mostly to defer the object/flow creation and have the possibility to wrap it in a class if needed. 

I.e. not relying on import side effects and module level variable that could potentially be mutated